### ISSUE TITLE
Implement details of sine activation

### ISSUE THREAD

**Issue by wzpscott** [NVlabs/tiny-cuda-nn#216]
Thanks for your nice paper and implementations.
Currently I try to use tiny-cuda-nn to speed up the training process of Siren (https://github.com/vsitzmann/siren). I use the following configs for my network：

~~~python
{
	"otype": "CutlassMLP",       
	"activation": "Sine",        
	"output_activation": "None", 
	"n_neurons": 256,            
	"n_hidden_layers": 3
}
~~~

However the network produce different results compared with the pytorch implementation of Siren. I suspect it is because Siren uses a special network initialization scheme. Can you clarify how tiny-cuda-nn is initialized with sine activation? 

The initialization scheme of Siren is something like:

```python
def init_weights(self):
    with torch.no_grad():
        if self.is_first_layer:
            self.linear.weight.uniform_(-1 / self.in_features, 1 / self.in_features)
        else:
            self.linear.weight.uniform_(
                -np.sqrt(6 / self.in_features) / self.omega_0,
                np.sqrt(6 / self.in_features) / self.omega_0,
            )
```

**Comment by Tom94** [comment]
Siren's initialization is implemented in __tiny-cuda-nn__ -- you can check the `initialize_params` method of `CutlassMLP` for reference.

One aspect that might cause the difference is that __tiny-cuda-nn__ does not have biases.

**Comment by xilongzhou** [comment]
I observe the similar issue, when I run Sine network, I cannot encode the image well. 

I check the initialization in (https://github.com/NVlabs/tiny-cuda-nn/blob/327b3699d8aab6c694ec43b7ee0aeced57301bfb/include/tiny-cuda-nn/gpu_matrix.h#L296) and (https://github.com/NVlabs/tiny-cuda-nn/blob/327b3699d8aab6c694ec43b7ee0aeced57301bfb/include/tiny-cuda-nn/gpu_matrix.h#L313), we do not have omega=30 here and in the forward function (https://github.com/NVlabs/tiny-cuda-nn/blob/abb808d56bd021175a325bfa4cacc18a3555acee/include/tiny-cuda-nn/common_device.h#L172), omega is not applied here.

**Comment by Tom94** [comment]
See `GPUMatrix::initialize_siren_uniform_first` for omega=30 and `GPUMatrix::initialize_siren_uniform` for the initialization of other layers.

**Comment by Tom94** [comment]
No, the siren paper only multiplies the first layer by 30. (See sec 3.2)

If you look at the official code [over here](https://github.com/vsitzmann/siren/blob/master/modules.py), the activation function is indeed implemented as `torch.sin(30 * x)` for all layers. But if you check the initialization, all non-first layers are initialized with a division by 30 (rather than the first with a multiplication by thirty as in tiny-cuda-nn). So the net effect is identical to what tiny-cuda-nn does.
